backend/src/channel/channel.service.ts

```diff
@@ -240,6 +247,38 @@ export class ChannelService {
     };
   }
 
+  /**
+   * mute 하기
+   */
+  async muteUser(myId: number, channel: Channel, targetId: number): Promise<SuccessResponseDto> {
+    this.checkOperationAuthority(myId, channel, targetId);
+    await this.cacheManager.set(`mute-${targetId}`, true, MUTE_EXPIRES_IN);
+    return { message: 'mute 되었습니다.' };
+  }
+
+  /**
+   * kick 하기
+   */
+  async kickUser(myId: number, channel: Channel, targetId: number): Promise<SuccessResponseDto> {
+    this.checkOperationAuthority(myId, channel, targetId);
+    await this.cacheManager.del(`mute-${targetId}`);
+    this.channelGateway.emitChannel<UserId>(channel.id, 'kicked', { userId: targetId });
+    this.connectionGateway.leaveChannel(targetId, channel, undefined);
+    return { message: 'kick 되었습니다.' };
+  }
+
+  /**
+   * ban 하기
+   */
+  async banUser(myId: number, channel: Channel, targetId: number): Promise<SuccessResponseDto> {
+    this.checkOperationAuthority(myId, channel, targetId);
+    await this.cacheManager.del(`mute-${targetId}`);
+    channel.bannedUserIdList.push(targetId);
+    this.channelGateway.emitChannel<UserId>(channel.id, 'banned', { userId: targetId });
+    this.connectionGateway.leaveChannel(targetId, channel, undefined);
```

Comment:
요거 undefined 안 넣어줘도 돼용 ㅎㅎ
```suggestion
    this.connectionGateway.leaveChannel(targetId, channel);
```

Reply:
추가로 `connectionGateway.leaveChannel`에서 emit하는 부분 수정해서 다시 부탁드립니다아~~

Reply:
수정했습니다!